tests/e2e/json_marshal_test.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+	"testing"
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/davecgh/go-spew/spew"
+)
+
+func TestProposalUnmarshal(t *testing.T) {
```

Comment:
Is there a reason these tests don't use rapid like `action_rapid_test.go` does?

Reply:
Yes. I wanted to expliticly give a json string as an input, since this is hard to do with Rapid - we can get a json input by marshalling the rapid-generated objects, but this implies trusting that the marshal function does things correctly. this is able to test unmarshal and marshal in a slightly more isolated manner